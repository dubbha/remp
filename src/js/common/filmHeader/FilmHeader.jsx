import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../logo';
import FilmButton from './FilmButton';
import ErrorBoundary from '../errorBoundary';
import filmPropShape from '../utils/propShapes';
import { imgUrl } from '../config/api.config';
import parseYear from '../utils/parseYear';
import mapGenres from '../utils/mapGenres';
import displayCast from '../utils/displayCast';
import './style.sass';

const FilmHeader = ({ film, onSearchClick }) => (
  <header className="film">
    <ErrorBoundary>
      <div className="film__headerContainer">
        <Logo />
        <FilmButton onSearchClick={onSearchClick} />
      </div>
      <div className="film__infoSection">
        <div className="film__poster">
          <img
            className="film__image"
            src={`${imgUrl}${film.poster_path}`}
            alt={film.title}
          />
        </div>
        <div className="film__details">
          <div className="film__title">{film.title}<div className="film__rating">{film.vote_average}</div></div>
          <div className="film__category">{mapGenres(film.genre_ids)}</div>
          <div className="film__year">{parseYear(film.release_date)}</div>
          <div className="film__runtime">{film.runtime && `${film.runtime}  min`}</div>
          <div className="film__summary">{film.overview}</div>
          <div className="film__director">{film.director && `Director: ${film.director}`}</div>
          <div className="film__cast">{displayCast(film.cast)}</div>
        </div>
      </div>
    </ErrorBoundary>
  </header>
);

FilmHeader.propTypes = {
  film: PropTypes.shape(filmPropShape).isRequired,
  onSearchClick: PropTypes.func.isRequired,
};

export default FilmHeader;
