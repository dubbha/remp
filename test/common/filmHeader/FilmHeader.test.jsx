import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FilmHeader from 'common/filmHeader';

describe('FilmHeader', () => {
  it('should render successfully', () => {
    const props = {
      onSearchClick: jest.fn(),
      film: {
        id: 12345678,
        title: 'TITLE',
        director: 'DIRECTOR',
        overview: 'OVERVIEW',
        poster_path: 'POSTER_PATH',
        release_date: '2000/12/31',
        runtime: 120,
        vote_average: 5.0,
        genre_ids: [1, 2, 3],
        cast: [
          { name: 'NAME1' },
          { name: 'NAME2' },
          { name: 'NAME3' },
        ],
      },
    };
    const wrapper = shallow(<FilmHeader {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
