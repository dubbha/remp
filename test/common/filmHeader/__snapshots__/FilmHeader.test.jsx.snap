// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmHeader should render successfully 1`] = `
<header
  className="film"
>
  <ErrorBoundary>
    <div
      className="film__headerContainer"
    >
      <Logo />
      <FilmButton
        onSearchClick={[Function]}
      />
    </div>
    <div
      className="film__infoSection"
    >
      <div
        className="film__poster"
      >
        <img
          alt="TITLE"
          className="film__image"
          src="https://image.tmdb.org/t/p/w500/POSTER_PATH"
        />
      </div>
      <div
        className="film__details"
      >
        <div
          className="film__title"
        >
          TITLE
          <div
            className="film__rating"
          >
            5
          </div>
        </div>
        <div
          className="film__category"
        >
          , , 
        </div>
        <div
          className="film__year"
        >
          2000
        </div>
        <div
          className="film__runtime"
        >
          120  min
        </div>
        <div
          className="film__summary"
        >
          OVERVIEW
        </div>
        <div
          className="film__director"
        >
          Director: DIRECTOR
        </div>
        <div
          className="film__cast"
        >
          Cast: NAME1, NAME2, NAME3
        </div>
      </div>
    </div>
  </ErrorBoundary>
</header>
`;
